Tests/SentryTests/Integrations/SessionReplay/SentryReplayOptionsTests.swift

```diff
@@ -3,34 +3,217 @@ import Foundation
 import XCTest
 
 class SentryReplayOptionsTests: XCTestCase {
+    // MARK: - Initializer
 
-    func testQualityConversion() {
+    func testInit_withoutArguments_shouldUseDefaults() {
+        // -- Act --
+        let options = SentryReplayOptions()
+
+        // -- Assert --
+        XCTAssertEqual(options.sessionSampleRate, 0)
+        XCTAssertEqual(options.onErrorSampleRate, 0)
+        XCTAssertTrue(options.maskAllText)
+        XCTAssertTrue(options.maskAllImages)
+        XCTAssertTrue(options.enableViewRendererV2)
+        XCTAssertFalse(options.enableFastViewRendering)
+
+        XCTAssertEqual(options.maskedViewClasses.count, 0)
+        XCTAssertEqual(options.unmaskedViewClasses.count, 0)
+        XCTAssertEqual(options.quality, SentryReplayOptions.SentryReplayQuality.defaultQuality)
+        XCTAssertEqual(options.frameRate, 1)
+        XCTAssertEqual(options.errorReplayDuration, 30)
+        XCTAssertEqual(options.sessionSegmentDuration, 5)
+        XCTAssertEqual(options.maximumDuration, 60 * 60)
+    }
+
+    func testInit_withAllArguments_shouldSetValues() {
+        // -- Act --
+        // Use the opposite of the default values to check if they are set correctly
+        let options = SentryReplayOptions(
+            sessionSampleRate: 0.5,
+            onErrorSampleRate: 0.8,
+            maskAllText: false,
+            maskAllImages: false,
+            enableViewRendererV2: false,
+            enableFastViewRendering: true
+        )
+
+        // -- Assert --
+        XCTAssertEqual(options.sessionSampleRate, 0.5)
+        XCTAssertEqual(options.onErrorSampleRate, 0.8)
+        XCTAssertFalse(options.maskAllText)
+        XCTAssertFalse(options.maskAllImages)
+        XCTAssertFalse(options.enableViewRendererV2)
+        XCTAssertTrue(options.enableFastViewRendering)
+
+        XCTAssertEqual(options.maskedViewClasses.count, 0)
+        XCTAssertEqual(options.unmaskedViewClasses.count, 0)
+        XCTAssertEqual(options.quality, SentryReplayOptions.SentryReplayQuality.defaultQuality)
+        XCTAssertEqual(options.frameRate, 1)
+        XCTAssertEqual(options.errorReplayDuration, 30)
+        XCTAssertEqual(options.sessionSegmentDuration, 5)
+        XCTAssertEqual(options.maximumDuration, 60 * 60)
+    }
+
+    func testInit_sessionSampleRateOmitted_shouldUseDefaultValues() {
```

Comment:
`l`: Can't we reduce the test cases by omitting all except one parameter in one test, and then omitting just this parameter in another test. We basically just need two test cases. So basically something like this

Test omitting all other than `sessionSampleRate`
```
let options = SentryReplayOptions(
            sessionSampleRate: 0.5,
        )
```

Tests omitting `sessionSampleRate`
```
let options = SentryReplayOptions(
           
            onErrorSampleRate: 0.8,
            maskAllText: false,
            maskAllImages: false,
            enableViewRendererV2: false,
            enableFastViewRendering: true
        )
```

Reply:
I believe we should test them omitted individually because removing most of the defaults could easily cause a different initializer being tested instead, e.g. removing all would cause the default initializer to be triggered instead of using all default values --> the original problem.

Relying on at least one value being set could work, which was the previous attempt aswell.

At least I can remove the assertions for unrelated fields, so we only assert the default value of each omitted field

Reply:
Let's keep them then.